src/components/collection/Picker.tsx

```diff
@@ -114,30 +120,63 @@ function CollectionPicker({ readOnly = false }: Props) {
                           }))
                     : [];
 
-            const draftSpecCollections: string[] = draftSpecCollectionData.map(
-                (collection) => collection.name
-            );
+            const draftSpecCollections: string[] =
+                draftSpecCollectionOptions.map((collection) => collection.name);
 
             const collectionsOnServer: CollectionData[] = [
-                ...draftSpecCollectionData,
-                ...liveSpecCollectionData.filter(
+                ...draftSpecCollectionOptions,
+                ...liveSpecCollectionOptions.filter(
                     ({ name }) => !draftSpecCollections.includes(name)
                 ),
             ];
 
-            setCollectionData(collectionsOnServer);
+            setCollectionOptions(collectionsOnServer);
         }
     }, [
-        setCollectionData,
+        setCollectionOptions,
         discoveredCollectionsLabel,
         draftSpecs,
         entityType,
         existingCollectionsLabel,
         liveSpecs,
-        populateCollectionData,
+        populateCollectionOptions,
         workflow,
     ]);
 
+    const populateCollectionValues = useMemo(() => {
+        return collections?.every((collection) =>
+            collectionOptions.find(({ name }) => name === collection)
+        );
+    }, [collections, collectionOptions]);
+
+    useEffect(() => {
+        if (populateCollectionValues && collections) {
+            const values = collections.map(
+                (collection) =>
+                    collectionOptions.find(
+                        ({ name }) => name === collection
+                    ) ?? {
+                        name: collection,
```

Comment:
The condition that sets `populateCollectionValues` should guarantee that there is an autocomplete option for each collection in the `collections` array, therefore the right-hand side of the nullish coalescing operator should never be reached. This assertion has proven to be true in my rounds of testing.